### ISSUE TITLE
🐛 [cloud_firestore_platform_interface] missing method FallThroughError

### ISSUE THREAD

**Issue by mooseandpebs** [firebase/flutterfire#10140]
## Bug report

when building with cloud_firestore_platform_interface 5.9.1 the error:
cloud_firestore_platform_interface-5.9.1/lib/src/platform_interface/utils/load_bundle_task_state.dart:17:13: Error: Method not found: 'FallThroughError'.
      throw FallThroughError();
            

### Steps to reproduce

1. add:  firebase_core: ^2.4.0
  firebase_storage: ^11.0.8
  cloud_firestore_platform_interface: ^5.9.1
  cloud_firestore: ^4.2.0
  firebase_auth: ^4.2.1
to pubspec.yaml
2. upgrade dependencies
3. flutter clean
4. flutter run 
5. wilburflutter >flutter --version
Flutter 3.7.0-10.0.pre.42 • channel master • https://github.com/flutter/flutter.git
Framework • revision d88d524050 (4 hours ago) • 2022-12-18 09:53:39 -0500
Engine • revision 276327f7ea
Tools • Dart 3.0.0 (build 3.0.0-35.0.dev) • DevTools 2.20.0

[bugs.txt](https://github.com/firebase/flutterfire/files/10254455/bugs.txt)

### Sample project

Providing a minimal example project which demonstrates the bug in isolation from your main App _greatly_ enhances the chance of a timely fix.
Please link to the public repository URL.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

**Comment by sma** [comment]
I get the same error, both for iOS, macOS and the web.  So it's completely broken at the moment. 

It looks like that this [commit](https://github.com/firebase/flutterfire/commit/0592d02ba036f7abb347e305aff52fdbfd568a56) isn't yet part of any release. That exception no [longer exists in Dart 3.0](https://github.com/dart-lang/sdk/commit/bcbf24ec3dd5772f60df185b4265f1bce00f4f2b) alpha and was [deprecated in Dart 2.x in October](https://github.com/dart-lang/sdk/commit/c13676f2b7bc52951a770731e602e50426101cf8).

**Comment by redtoothmc** [comment]
I am getting the same error as well...

**Comment by teimusheru** [comment]
I am getting the same error as well...

**Comment by baek-b** [comment]
Agree with @sma on the commit issue. 

As a temporary measure to not lose time if you are getting the FallThroughError. 
cmd+click on the file name and change the last bit (line 17) like so: 

`throw FallThroughError();`

to 

`throw UnsupportedError('Unknown LoadBundleTaskState value: $state.'); `

This is based off the updated [commit](https://github.com/firebase/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/platform_interface/utils/load_bundle_task_state.dart). My app runs after this change. Keep in mind that if you clean your pub cache - you will lose this change. This is just temporary.

**Comment by darshankawar** [comment]
Thanks for the report. I am able to replicate the error using a sample app, adding the said dependencies:

```
dhs@Dhss-MacBook-Pro firebase_core_example % flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
../../../../../.pub-cache/hosted/pub.dev/cloud_firestore_web-3.1.1/lib/src/utils/web_utils.dart:71:13: Error: Method not found: 'FallThroughError'.
      throw FallThroughError();
            ^^^^^^^^^^^^^^^^
../../../../../.pub-cache/hosted/pub.dev/cloud_firestore_platform_interface-5.9.1/lib/src/platform_interface/utils/load_bundle_task_state.dart:17:13: Error: Method not found:
'FallThroughError'.
      throw FallThroughError();
            ^^^^^^^^^^^^^^^^
../../../../../.pub-cache/hosted/pub.dev/firebase_auth_platform_interface-6.11.3/lib/src/action_code_info.dart:65:15: Error: The method 'FallThroughError' isn't defined for the class
'ActionCodeInfo'.
 - 'ActionCodeInfo' is from 'package:firebase_auth_platform_interface/src/action_code_info.dart'
 ('../../../../../.pub-cache/hosted/pub.dev/firebase_auth_platform_interface-6.11.3/lib/src/action_code_info.dart').
Try correcting the name to the name of an existing method, or defining a method named 'FallThroughError'.
        throw FallThroughError();
              ^^^^^^^^^^^^^^^^
Waiting for connection from debug service on Chrome...             24.4s
Failed to compile application.
dhs@Dhss-MacBook-Pro firebase_core_example % 

```

**Comment by Lyokone** [comment]
I just ran an app without any issues on Web and iOS with
```
  cloud_firestore: ^4.2.0
  firebase_auth: ^4.2.1
  firebase_core: ^2.4.0
  firebase_storage: ^11.0.8
```
(you shouldn't manually add the platform_interface).

Can you try to run `flutterfire update` to see if your lock file has been properly updated?

**Comment by sma** [comment]
As long as this configuration still uses `cloud_firestore_platform_interface-5.9.1`, the problem persists. It refers to a deleted class called `FallThroughError`.  Did you use the `master` channel to reproduce the problem?

**Comment by Lyokone** [comment]
Oh, I didn't use master, only stable. It appears that `cloud_firestore_platform_interface` wasn't properly deployed last week. I'm checking that.

**Comment by Lyokone** [comment]
I'm doing a manual release for the fix that wasn't deployed, I'll let you know when it's up: https://github.com/firebase/flutterfire/pull/10141

**Comment by sma** [comment]
Well. It's working on macOS and iOS again. Thanks!

But for the web, you need to fix `cloud_firestore_web-3.1.1` which still refers to `FallThroughError`, I'm afraid.

**Comment by kraa** [comment]
Yeah, cloud_firestore_web still use FallThroughError

`cloud_firestore_web-3.1.1/lib/src/utils/web_utils.dart:71:13:
Error: Method not found: 'FallThroughError'.
      throw FallThroughError();
            ^^^^^^^^^^^^^^^^
Error: Compilation failed.
`

**Comment by Lyokone** [comment]
Ah I missed that this package was also not correctly released. You can use `master` version till Thursday, when we'll do a proper release 👍

**Comment by SubodhLandmark** [comment]
I am facing error. Any quick help will appreciated. My running build broke.
cloud_firestore_web-3.1.1/lib/src/utils/web_utils.dart:71:13: Error: Method not found: 'FallThroughError'. throw FallThroughError(); ^^^^^^^^^^^^^^^^ Error: Compilation failed.

**Comment by dwivedikunal100** [comment]
I try to use it from git and get this error.


```
Because _flutterfire_internals 1.0.10 depends on cloud_firestore_web ^3.1.1 and no versions of _flutterfire_internals match >1.0.10 <2.0.0, _flutterfire_internals ^1.0.10 requires cloud_firestore_web from hosted.
So, because app depends on cloud_firestore_web from git which depends on _flutterfire_internals ^1.0.10, version solving failed.
```

In pubsec.yaml
```
  cloud_firestore_web:
    git:
      url: https://github.com/firebase/flutterfire.git
      ref: master # branch name
      path: packages/cloud_firestore/cloud_firestore_web
```

**Comment by Albejr** [comment]
> Agree with @sma on the commit issue.
> 
> As a temporary measure to not lose time if you are getting the FallThroughError. cmd+click on the file name in the error(load_bundle_tast_state.dart) and change the last bit (line 17) like so:
> 
> `throw FallThroughError();`
> 
> to
> 
> `throw UnsupportedError('Unknown LoadBundleTaskState value: $state.'); `
> 
> This is based off the updated [commit](https://github.com/firebase/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/platform_interface/utils/load_bundle_task_state.dart). My app runs after this change. Keep in mind that if you clean your pub cache - you will lose this change. You could override the dependency if you really want to but I hope this gets fixed soon.

throw UnsupportedError('Unknown LoadBundleTaskState value: $changeType.');

**Comment by Lyokone** [comment]
We have released [cloud_firestore - v4.3.0](https://github.com/firebase/flutterfire/blob/master/CHANGELOG.md#cloud_firestore---v430) that contains the fixes. Feel free to reopen if you still have any issue.
